graphsync.go

```diff
@@ -137,6 +137,13 @@ func (e RequestCancelledErr) Error() string {
 	return "Request Failed - Responder Cancelled"
 }
 
+// RequestNotFoundErr indicates that a request with a particular request ID was not found
+type RequestNotFoundErr struct{}
+
+func (e RequestNotFoundErr) Error() string {
+	return "request not found"
+}
+
```

Comment:
I'm curious why a struct type instead of an exported var, similar to `ErrExtensionAlreadyRegistered`?

Reply:
I had the same question yesterday :)
https://github.com/ipfs/go-graphsync/pull/182#discussion_r682480610

Reply:
Interesting; thank you

Reply:
yea sorry for my paranoia. I almost never use exported var errors.

Reply:
TIL; thank you both. [This](https://dave.cheney.net/tag/errors) is a great read; thanks @hannahhoward